### PR TITLE
quality of life improvements + repo bootstrapping

### PR DESCRIPTION
**Pull request by ashilen**
**Overview**
##### Logging
Introducing a logger to be used instead of `print`, so that
a) log streams can be manipulated more easily in the future should we want to, for instance, send them to sentry when sykle is used in build environments, and so that
b) users feel like it's slick. The `FancyLogger` class converts messages to colors according to their level, so `critical` is red, `warning` is yellow, `info` green, etc.. `FancyLogger` also exposes a `halo` method that can be used as a context manager or function decorator, which captures the log stream and prints it, in place, next to a fancy spinner. `halo` has an optional argument, `success`, that indicates if a `✔Success!` message should be the last thing the user sees once their command completes.

##### SCM plugin
The SCM plugin does (most of) the work that we usually do by hand when we make a new repo. It
1. Creates the repo.
2. Makes a default `develop` branch.
3. Adds the `typecode-developers` and `bots` teams to the repo, with admin access.
4. Adds protections to the `develop` and `master` branches. Note that these protections are slightly more restrictive than our usual protections. They require at least one approving review before a PR may be merged.
5. Adds a webhook to `smash.typecode.de`.
6. Clones the repo into the current working directory.

Future items for development of the SCM plugin should be: 
1. Parameterization of the defaults in `plugins.scm.scm.SCM`, for instance the `smash` url and the `default_team`, so that they can be overridden from `plugins.scm.Plugin`.
2. Addition of a default jira webhook and a corresponding parameter for the cli.

##### Plugin installation
The SCM plugin has its own dependencies. I think it makes sense (obv open to discussion) to keep its dependencies in its own folder. To that end I've made some changes to the `plugin_utils` module to support discovery and installation of plugin-specific dependencies, added a `plugins install` command, and made the previous `plugins` command more explicitly a list command.

### CODE REVIEW

**Review by LevKanter — APPROVED** [review]
Looks great! Just want @Didericis's review and signoff before we merge.

Also, super nitpicky but I think it makes more sense to call this version 0.6.0 (since this brings in a lot of new functionality in a backwards compatible manner).

**Comment by Didericis** [comment]
@ashilen Ditto on bumping to 0.6.0. Love the changes, think keeping plugin dependencies separate was needed/def a good call. Feel free to merge once release number gets changed.

**Review by Didericis — APPROVED** [review]

**Pull request merged**
